localstack/services/events/provider.py

```diff
@@ -565,14 +565,23 @@ def events_handler_put_events(self):
         if not matching_rules:
             continue
 
+        event_time = datetime.datetime.utcnow()
+        if event_timestamp := event.get("Time"):
+            try:
+                # if provided, use the time from event
+                event_time = datetime.datetime.utcfromtimestamp(event_timestamp)
+            except ValueError:
+                # if we can't parse it, pass and keep using `utcnow`
```

Comment:
1. is this in parity with AWS or a simplification we do for LocalStack?
2. I'd recommend at least logging in this case, since users might wonder why their custom provided event time isn't working.

Reply:
1. good question, I think boto is validating this field so it's a bit tricky to validate with tests, and most SDK should
2. I'll add the logging, if it happens this is an issue, we can try to make a more complex test case. I didn't want to introduce a breaking change, so I fell back to the previous behaviour.